dainemo/autograd/ops/pool.mojo

```diff
@@ -0,0 +1,146 @@
+from tensor import Tensor, TensorShape
+from math import floor, min , max
+
+from dainemo import GRAPH
+from dainemo.autograd.node import Node
+from dainemo.autograd.ops.conv import get_result_shape
+from dainemo.utils.tensorutils import calculate_strides
+
+
+# <------------MAXPOOL2D------------>
+struct MAXPOOL2D:
+    @staticmethod
+    fn forward[
+        kernel_shape: TensorShape,
+        padding: StaticIntTuple[2] = 0,
+        stride: StaticIntTuple[2] = 1,
+        dilation: StaticIntTuple[2] = 1
+    ](inputs: Node[dtype]) -> Node[dtype]:
+        """
+        Returns the max value of each kernel in the input tensor.
+            inputs.shape     [batch_size, in_channels, iX, iY]
+            kernel.shape     [out_channels, in_channels, kX, kY] 
+            outputs.shape    [batch_size, out_channels, oX, oY]
+            and for maxpool2d (in_channels == out_channels).
+        """
+
+        alias nelts: Int = simdwidthof[dtype]()
+        
+        let result_shape = get_result_shape[padding, stride, dilation](
+            inputs.tensor.shape(), kernel_shape
+        )
+
+        var outputs = Tensor[dtype](
+            inputs.tensor.dim(0), kernel_shape[0], result_shape[0], result_shape[1]
+        )
+        let outputs_strides = calculate_strides(outputs.shape())
+
+        for batch in range(inputs.tensor.dim(0)):
+            for in_ch in range(inputs.tensor.dim(1)):
+                for x in range(outputs.dim(2)):
+                    for y in range(outputs.dim(3)):
+                        var max_val: SIMD[dtype, 1] = -1e9
```

Comment:
I think this could be changed with math.limit.min_finite or math.limit.neginf in the forward and backward pass, to be able to also represent the biggest value for any dtype. And also why -1e9, isn't the biggest value for 32bits (approx 1.1 x 10^-38), or are we never going to have a smaller value than -1e9 in maxpool?

Reply:
Interesting, didn't know about these. Will fix it